Library/Homebrew/brew.sh

```diff
@@ -433,12 +433,14 @@ then
   # shellcheck disable=SC2086,SC2183
   printf -v HOMEBREW_MACOS_VERSION_NUMERIC "%02d%02d%02d" ${HOMEBREW_MACOS_VERSION//./ }
 
-  # Don't include minor versions for Big Sur and later.
+  # Don't include minor versions for Big Sur and later, or patch versions for Catalina and older.
   if [[ "${HOMEBREW_MACOS_VERSION_NUMERIC}" -gt "110000" ]]
   then
-    HOMEBREW_OS_VERSION="macOS ${HOMEBREW_MACOS_VERSION%.*}"
+    HOMEBREW_MACOS_VERSION_NUMBER="${HOMEBREW_MACOS_VERSION%%.*}"
+    HOMEBREW_OS_VERSION="macOS ${HOMEBREW_MACOS_VERSION_NUMBER}"
   else
-    HOMEBREW_OS_VERSION="macOS ${HOMEBREW_MACOS_VERSION}"
+    HOMEBREW_MACOS_VERSION_NUMBER="${HOMEBREW_MACOS_VERSION%.*}"
```

Comment:
The name `HOMEBREW_MACOS_VERSION_NUMBER` feels too similar to `HOMEBREW_MACOS_VERSION_NUMERIC`.

This does slightly change how analytics work but I don't know if we actually have full access to patch versions there anyway or not (and if such data is useful or not).

Reply:
I can also just skip defining `HOMEBREW_MACOS_VERSION_NUMBER` and strip out the `macOS ` from `HOMEBREW_OS_VERSION`. Would that be better?

Reply:
I honestly see no reason to strip patch information from `HOMEBREW_OS_VERSION` so it really could just be a constant `"macOS ${HOMEBREW_MACOS_VERSION}"` for that.

Reply:
Yeah it was analytics (https://github.com/Homebrew/brew/commit/8af489547945bd9fb3b42c9e99123ca5d1b0ec2e) so it should indeed no longer be an issue since https://github.com/Homebrew/homebrew-formula-analytics/commit/3be112196eb44e2bfca7fbbf7c79cb0f2028017d.

Feel free to simplify and remove the special casing.

Reply:
I've removed the special casing from generating `HOMEBREW_OS_VERSION`, but we still need it to find the right `pkgconfig` directory.

I've also changed `HOMEBREW_MACOS_VERSION_NUMBER`.